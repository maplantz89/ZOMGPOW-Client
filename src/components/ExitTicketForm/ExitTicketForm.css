.modal {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -30%);
  background-color: white;
  padding: 1rem 1.5rem;
  width: 80%;
  border-radius: 15px;
  border: 1px solid darkgray;
}

.modal-container {
  position: fixed;
  height: 100vh;
  width: 100vw;
  top: 0;
  left: 0;
  background: rgba(255, 255, 255, 0.7);
  overflow-y: scroll
}

.close {
  border-radius: 3px;
  float: right;
  width: 15px;
  color: var(--red);
  font-weight: 800;
  font-size: 20px;
}

.close:hover {
  color: black;
  cursor: pointer;
}

.modal textarea {
  width: 80%;
}

.exit-ticket-button-container {
  display: flex;
  flex-direction: column;
}
.exit-ticket-button-container button {
  margin-bottom: 5px;
}

/* MEDIA QUERIES */

@media only screen and (min-width: 600px) {
  .modal {
    transform: translate(-50%, -50%);
    width: 500px;
  }

  .exit-ticket-button-container button {
    width: 50%;
    margin: 0 auto;
    margin-bottom: 5px;
  }
}
