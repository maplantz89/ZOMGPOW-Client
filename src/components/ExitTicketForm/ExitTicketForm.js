import React from 'react';
import './ExitTicketForm.css';

class ExitTicketForm extends React.Component {

  state = {
    error: null,
    multipleChoiceVisible: false,
    shortAnswerVisible: false,
  }

  multipleChoice = () => {
    this.setState({multipleChoiceVisible: true, shortAnswerVisible: false})
  }

  shortAnswer = () => {
    this.setState({multipleChoiceVisible: false, shortAnswerVisible: true})
  }

  onClose = () => {
    this.setState({multipleChoiceVisible: false, shortAnswerVisible: false});
    this.props.onClose();
  }

  render() {
    if(!this.props.show) {
      return null;
    }
    return (
      <div className='modal-container'>
      <div className='modal'>
        <div className='close' onClick={this.onClose}>X</div>
        <h2>Would you like to create an exit ticket for this learning target?</h2>
        <div className={this.state.multipleChoiceVisible ? '' : 'hidden'}>
          <form 
            className='form multiple-choice'
            onSubmit={(e) => this.props.submitExitTicket(e, 'multipleChoice')}
            >
            <label htmlFor='exit-ticket-question-mc'>Exit Ticket Question:</label>
            <textarea 
              id='exit-ticket-question-mc' 
              onChange={(e) => this.props.updateQuestion(e)}
              value={this.props.exitTicketQuestion}
              required
              />
            <label htmlFor='option-A'>Option A:</label>
            <input 
              id='option-A' 
              onChange={(e) => this.props.updateMultipleChoiceSubmit(e)}
              value={this.props.multipleChoiceA}
              required
            />
            <label htmlFor='option-B'>Option B:</label>
            <input 
              id='option-B'
              onChange={(e) => this.props.updateMultipleChoiceSubmit(e)}
              value={this.props.multipleChoiceB} 
              required
            />
            <label htmlFor='option-C'>Option C:</label>
            <input 
              id='option-C' 
              onChange={(e) => this.props.updateMultipleChoiceSubmit(e)}
              value={this.props.multipleChoiceC}
              required
            />
            <label htmlFor='option-D'>Option D:</label>
            <input 
              id='option-D' 
              onChange={(e) => this.props.updateMultipleChoiceSubmit(e)}
              value={this.props.multipleChoiceD}
              required
            />
            <label htmlFor='exit-ticket-correct-answer'>Correct Option:</label>
            <div>
              <input 
                type='radio' 
                name='radio'
                id='correct-option-A'
                value='A'
                onChange={(e) => this.props.updateCorrectAnswer(e, 'A')}
                />
              <label 
                htmlFor='correct-option-A'
              >A</label>

              <input 
                type='radio' 
                name='radio'
                id='option-B'
                value='B'
                onChange={(e) => this.props.updateCorrectAnswer(e, 'B')}              />
              <label 
                htmlFor='option-B'
              >B</label>

              <input 
                type='radio'
                name='radio' 
                id='option-C'
                value='C'
                onChange={(e) => this.props.updateCorrectAnswer(e, 'C')}              />
              <label 
                htmlFor='option-C'
              >C</label>

              <input 
                type='radio' 
                name='radio'
                id='option-D'
                value='D'
                onChange={(e) => this.props.updateCorrectAnswer(e, 'D')}              />
              <label 
                htmlFor='option-D'
              >D</label>
            </div>
              {this.state.error && <p className='alert'>{this.state.error}</p>}
            <button 
              type='submit'
              className='button green-button'
            >Start Session</button>
          </form>
        </div>
        <div className={this.state.shortAnswerVisible ? '' : 'hidden'}>
          <form 
            className='form short-answer'
            onSubmit={(e) => this.props.submitExitTicket(e, 'shortAnswer')}
            >
            <label htmlFor='exit-ticket-question'>Exit Ticket Question:</label>
            <textarea 
              id='exit-ticket-question-sa' 
              onChange={(e) => this.props.updateQuestion(e)}
              value={this.props.exitTicketQuestion}
              required
            />
            <button 
              type='submit'
              className='button green-button'
            >Start Session</button>
          </form>
          </div>
        <div className='exit-ticket-button-container'>
          <button 
            className='button blue-button'
            onClick={() => this.setState({multipleChoiceVisible: true, shortAnswerVisible: false})}
          >Multiple Choice
          </button>
          <button 
            className='button purple-button'
            onClick={() => this.setState({multipleChoiceVisible: false, shortAnswerVisible: true})}
          >Short Answer
          </button>
          <button 
            className='button green-button'
            onClick={(e) => this.props.handleSubmit(e)}
          >No thanks, start my session
          </button>
        </div>
      </div>

        
        
      </div>
    )
  }
}

export default ExitTicketForm;