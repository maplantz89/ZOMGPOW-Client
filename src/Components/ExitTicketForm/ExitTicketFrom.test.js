import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ExitTicketForm from './ExitTicketForm';

describe(`ExitTicketForm component`, () => {
  it('renders a .ExitTicketForm by default', () => {
    const wrapper = shallow(<ExitTicketForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})