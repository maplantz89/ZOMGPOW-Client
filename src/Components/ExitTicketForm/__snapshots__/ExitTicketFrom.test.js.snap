// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExitTicketForm component renders a .ExitTicketForm by default 1`] = `
<div
  className="exit-ticket-form"
>
  <p>
     -OR- 
  </p>
  <button
    className="button purple-button"
    onClick={[Function]}
  >
    Create an Exit Ticket for this Learning Target?
  </button>
  <div
    className="hidden"
  >
    <button
      className="button blue-button"
      onClick={[Function]}
    >
      Multiple Choice
    </button>
    <button
      className="button green-button"
      onClick={[Function]}
    >
      Short Answer
    </button>
  </div>
  <div
    className="hidden"
  >
    <form
      className="form multiple-choice"
      onSubmit={[Function]}
    >
      <label
        htmlFor="exit-ticket-question-mc"
      >
        Exit Ticket Question:
      </label>
      <textarea
        id="exit-ticket-question-mc"
        onChange={[Function]}
        required={true}
        value=""
      />
      <label
        htmlFor="option-A"
      >
        Option A:
      </label>
      <input
        id="option-A"
        onChange={[Function]}
        required={true}
        value=""
      />
      <label
        htmlFor="option-B"
      >
        Option B:
      </label>
      <input
        id="option-B"
        onChange={[Function]}
        required={true}
        value=""
      />
      <label
        htmlFor="option-C"
      >
        Option C:
      </label>
      <input
        id="option-C"
        onChange={[Function]}
        required={true}
        value=""
      />
      <label
        htmlFor="option-D"
      >
        Option D:
      </label>
      <input
        id="option-D"
        onChange={[Function]}
        required={true}
        value=""
      />
      <label
        htmlFor="exit-ticket-correct-answer"
      >
        Correct Option:
      </label>
      <div>
        <input
          id="correct-option-A"
          name="radio"
          onChange={[Function]}
          type="radio"
          value="A"
        />
        <label
          htmlFor="correct-option-A"
        >
          A
        </label>
        <input
          id="option-B"
          name="radio"
          onChange={[Function]}
          type="radio"
          value="B"
        />
        <label
          htmlFor="option-B"
        >
          B
        </label>
        <input
          id="option-C"
          name="radio"
          onChange={[Function]}
          type="radio"
          value="C"
        />
        <label
          htmlFor="option-C"
        >
          C
        </label>
        <input
          id="option-D"
          name="radio"
          onChange={[Function]}
          type="radio"
          value="D"
        />
        <label
          htmlFor="option-D"
        >
          D
        </label>
      </div>
      <button
        className="button purple-button"
        type="submit"
      >
        Save Exit Ticket
      </button>
    </form>
  </div>
  <div
    className="hidden"
  >
    <form
      className="form short-answer"
      onSubmit={[Function]}
    >
      <label
        htmlFor="exit-ticket-question"
      >
        Exit Ticket Question:
      </label>
      <textarea
        id="exit-ticket-question-sa"
        onChange={[Function]}
        required={true}
        value=""
      />
      <button
        className="button purple-button"
        type="submit"
      >
        Save Exit Ticket
      </button>
    </form>
  </div>
</div>
`;
